ibm/resource_ibm_is_image.go

```diff
@@ -0,0 +1,330 @@
+			},
+
+			ResourceName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The name of the resource",
+			},
+
+			ResourceCRN: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The crn of the resource",
+			},
+
+			ResourceStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the resource",
+			},
+
+			ResourceGroupName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The resource group name in which resource is provisioned",
+			},
+		},
+	}
+}
+
+func resourceIBMISImageCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).ISSession()
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Image create")
+	href := d.Get(isImageHref).(string)
+	name := d.Get(isImageName).(string)
+	operatingSystem := d.Get(isImageOperatingSystem).(string)
+
+	imageC := compute.NewImageClient(sess)
+	image, err := imageC.Create(href, name, operatingSystem)
+	if err != nil {
+		log.Printf("[DEBUG] Image err %s", err)
+		return err
+	}
+
+	d.SetId(image.ID.String())
+	log.Printf("[INFO] Image : %s", image.ID.String())
+
+	_, err = isWaitForImageAvailable(imageC, d.Id(), d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return err
+	}
+
+	if _, ok := d.GetOk(isImageTags); ok {
+		oldList, newList := d.GetChange(isImageTags)
+		err = UpdateTagsUsingCRN(oldList, newList, meta, image.Crn)
+		if err != nil {
+			log.Printf(
+				"Error on create of resource image (%s) tags: %s", d.Id(), err)
+		}
+	}
+	return resourceIBMISImageRead(d, meta)
+}
+
+func isWaitForImageAvailable(imageC *compute.ImageClient, id string, timeout time.Duration) (interface{}, error) {
+	log.Printf("Waiting for image (%s) to be available.", id)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"retry", isImageProvisioning},
+		Target:     []string{isImageProvisioningDone},
+		Refresh:    isImageRefreshFunc(imageC, id),
+		Timeout:    timeout,
+		Delay:      10 * time.Second,
+		MinTimeout: 10 * time.Second,
+	}
+
+	return stateConf.WaitForState()
+}
+
+func isImageRefreshFunc(imageC *compute.ImageClient, id string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		image, err := imageC.Get(id)
+		if err != nil {
+			return nil, "", err
+		}
+
+		if image.Status == "available" || image.Status == "failed" {
+			return image, isImageProvisioningDone, nil
+		}
+
+		return image, isImageProvisioning, nil
+	}
+}
+
+func resourceIBMISImageUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	sess, err := meta.(ClientSession).ISSession()
+	if err != nil {
+		return err
+	}
+	imageC := compute.NewImageClient(sess)
+
+	if d.HasChange(isImageName) {
+		name := d.Get(isImageName).(string)
+		_, err := imageC.Update(d.Id(), name)
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
Can you add a logic to d.HasChange(isImageTags) ...because tags can be changed after creation